yb-voyager/src/queryparser/unsupportedConstructs.go

```diff
@@ -160,3 +166,147 @@ func containsAdvisoryLocksInNodeList(nodes []*pg_query.Node) bool {
 	}
 	return false
 }
+
+func (qp *QueryParser) containsSystemColumns() bool {
+	if qp.ParseTree == nil {
+		log.Infof("parse tree not available for query-%s", qp.QueryString)
+		return false
+	}
+
+	selectStmtNode, isSelectStmt := qp.ParseTree.Stmts[0].Stmt.Node.(*pg_query.Node_SelectStmt)
+	// Note: currently considering only SELECTs but I/U/D dmls are also possible
+	if !isSelectStmt {
+		return false
+	}
+
+	if containsSystemColumnsInTargetList(selectStmtNode.SelectStmt.TargetList) {
+		return true
+	}
+
+	if containsSystemColumnsInFromClause(selectStmtNode.SelectStmt.FromClause) {
+		return true
+	}
+
+	if containsSystemColumnsInWhereClause(selectStmtNode.SelectStmt.WhereClause) {
+		return true
+	}
+
+	return false
+}
+
+/*
+Query: SELECT xmin, xmax FROM employees
+ParseTree:	version:160001  stmts:{stmt:{select_stmt:{target_list:{res_target:{val:{column_ref:{fields:{string:{sval:"xmin"}}  location:7}}  location:7}}
+target_list:{res_target:{val:{column_ref:{fields:{string:{sval:"xmax"}}  location:13}}  location:13}}
+from_clause:{range_var:{relname:"employees"  inh:true  relpersistence:"p"  location:23}}  limit_option:LIMIT_OPTION_DEFAULT  op:SETOP_NONE}}}
+*/
+func containsSystemColumnsInTargetList(targetList []*pg_query.Node) bool {
+	for _, target := range targetList {
+		targetRes := target.GetResTarget()
+		if targetRes == nil {
+			continue
+		}
+
+		val := targetRes.GetVal()
+		if val == nil {
+			continue
+		}
+
+		columnRef := val.GetColumnRef()
+		if columnRef == nil {
+			continue
+		}
+
+		fields := columnRef.GetFields()
+		if len(fields) == 0 {
+			continue
+		}
+
+		lastField := fields[len(fields)-1]
+		if lastField == nil || lastField.GetString_() == nil {
+			continue
+		}
+
+		colName := lastField.GetString_().Sval
+		if slices.Contains(unsupportedSysCols, colName) {
+			return true
```

Comment:
We are returning from the first unsupported system column found in the target column list right?
shouldn't we be reporting all of them?

Reply:
We will be reporting queries under a category - System Columns.
Wont report the exact column names for now.

Can enhance this later by adding Notes/Details field

Reply:
Okay, got it